portal-bridge/src/bridge/history.rs

```diff
@@ -200,11 +215,27 @@ impl HistoryBridge {
                 let epoch_acc = epoch_acc.clone();
                 let portal_clients = self.portal_clients.clone();
                 let execution_api = self.execution_api.clone();
+                let permit = match gossip_send_semaphore.clone().acquire_owned().await {
+                    Ok(permit) => permit,
+                    Err(_) => {
+                        error!("acquire_owned() can only error on semaphore close, this should be impossible");
+                        continue;
+                    }
+                };
                 tokio::spawn(async move {
-                    let _ =
+                    if (timeout(
```

Comment:
I'm not sure if I'm wrong here, but it seems like there is some duplicated logic here & in the other location where we spawn the thread in `"latest"`. Seems like we should be able to move some of this logic to `Self::serve_full_block()` since we're doing the same thing in each location. It looks like `Self::serve_full_block()` could take care of everything from spawning the thread and everything inside the spawn (eg error handling & dropping the permit)

Reply:
We can't move the logic to ``Self::serve_full_block()`` because timeout() requires we pass in a future. So I created a non-async function called ``spawn_serve_full_block`` which I believe accomplishes the refactor you were looking for.